framework/include/controls/ControllableParameter.h

```diff
@@ -0,0 +1,157 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef CONTROLLABLEPARAMETER_H
+#define CONTROLLABLEPARAMETER_H
+
+// MOOSE includes
+#include "InputParameters.h"
+#include "MooseObjectName.h"
+#include "MooseObjectParameterName.h"
+
+/**
+ * A class for accessing controllable InputParameters.
+ *
+ * This class is used within Control objects for returning sets
+ * of InputParaemeters.
+ */
+template<typename T>
+class ControllableParameter
```

Comment:
I believe @aeslaughter went this way because I asked him to encapsulate all of the tag parsing into a class. I was mostly just concerned about the parsing in the first iteration of this PR all over the place. I didn't necessarily mean to build it right into the `ControllableParamemter` class though. It's possible to have to separate classes to accomplish this. One that just does all the parsing and set up of the tags, groups, etc. and then a simple iterface like @friedmud proposed with a utility method for setting the "tag" -> value pair.